### PR TITLE
python bindings for SceneNode bounding boxes

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

For domain randomization, it is often useful to sample a 3D position from inside a scene. This PR enables users to compute/query the approximate scene bounding box from the SceneGraph root node (or any other SceneNode of interest).

## How Has This Been Tested

added python test

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/bindings/SceneBindings.cpp [hunk 1]
```diff
@@ -75,6 +75,17 @@ void initSceneBindings(py::module& m) {
       .def(
           "create_child", [](SceneNode& self) { return &self.createChild(); },
           R"(Creates a child node, and sets its parent to the current node.)")
+      .def(
+          "compute_cumulative_bb",
```
Comment (Skylion007) [inline src/esp/bindings/SceneBindings.cpp]: Is this suppose to not have a return statement?

File: src/esp/bindings/SceneBindings.cpp [hunk 2]
```diff
@@ -75,6 +75,17 @@ void initSceneBindings(py::module& m) {
...
+          [](SceneNode& self) { self.computeCumulativeBB(); },
```
Comment (erikwijmans) [inline src/esp/bindings/SceneBindings.cpp]:
```suggestion
          &SceneNode::computeCumulativeBB,
```

File: src/esp/bindings/SceneBindings.cpp [hunk 3]
```diff
@@ -75,6 +75,17 @@ void initSceneBindings(py::module& m) {
...
+          R"(Recursively compute the approximate axis aligned bounding boxes of the SceneGraph sub-tree rooted at this node.)")
+      .def(
+          "get_cumulative_bb",
+          [](SceneNode& self) { return self.getCumulativeBB(); },
```
Comment (erikwijmans) [inline src/esp/bindings/SceneBindings.cpp]:
```suggestion
          &SceneNode::getCumulativeBB,
```

File: src/esp/bindings/SceneBindings.cpp [hunk 4]
```diff
@@ -75,6 +75,17 @@ void initSceneBindings(py::module& m) {
...
+          R"(Get the approximate axis aligned bounding box of the SceneGraph sub-tree rooted at this node.)")
+      .def(
+          "get_mesh_bb", [](SceneNode& self) { return self.getMeshBB(); },
```
Comment (erikwijmans) [inline src/esp/bindings/SceneBindings.cpp]:
```suggestion
         &SceneNode::getMeshBB,
```

**Pull request merged**